client/gui/CIntObject.cpp

```diff
@@ -314,12 +314,20 @@ CKeyShortcut::CKeyShortcut(EShortcut key)
 {
 }
 
+CKeyShortcut::CKeyShortcut(const EShortcut & key, const KeyPressedFunctor & keyPressedCallback)
+	: CKeyShortcut(key)
+{
+	this->keyPressedCallback = keyPressedCallback;
+}
+
 void CKeyShortcut::keyPressed(EShortcut key)
 {
 	if( assignedKey == key && assignedKey != EShortcut::NONE && !shortcutPressed)
 	{
 		shortcutPressed = true;
 		clickPressed(GH.getCursorPosition());
+		if(keyPressedCallback)
+			keyPressedCallback();
```

Comment:
Why not override clickPressed instead of callback?

Reply:
CArtifactsOfHeroMain does not inherit from CKeyShortcut. CArtifactsOfHeroMain contains a vector of handlers for each key. That's why a callback is needed.

Reply:
But it is possible to do it through inheritance. Perhaps this option is better

Reply:
Inheritance is probably more clear.
Callback can be set by any entity from anywhere (and pain to debug since debugger loves going into std::function internals). And in case of inheritance list of functions that can be called is way easier to find.